File: API/Helpers/UserParams.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace API.Helpers
{
    public class UserParams
    {
        private const int MAX_PAGE_SIZE = 50;
        public int PageNumber { get; set; } = 1;
        private int _pageSize = 10;
        public int PageSize
        {
            get => _pageSize;
            set => _pageSize = (MAX_PAGE_SIZE > value) ? value : MAX_PAGE_SIZE;
        }
        public string Gender { get; set; }
        public string CurrentUserName { get; set; }

        public int MinAge { get; set; } = 18;
        public int MaxAge { get; set; } = 150;
        public string OrderBy { get; set; } = "lastactive";
    }
}

File: client/src/app/_models/userParams.ts
import { User } from "./user";

export class UserParams{
    gender: string;
    minAge: number = 18;
    maxAge: number = 99;
    pageNumber: number = 1;
    pageSize: number = 3;
    orderBy: string = 'lastActive';

    constructor(currentUser: User){
        if(currentUser.gender === 'male'){
            this.gender = 'female';
        }
        else{
            this.gender = 'male';
        }

    }
    setPageParam(pn:number, ps:number){
        this.pageSize=ps;
        this.pageNumber=pn;
    }
}
